python/cudf/cudf/tests/test_parquet.py

```diff
@@ -3768,6 +3788,42 @@ def test_parquet_chunked_reader(
     assert_eq(expected, actual)
 
 
+@pytest.mark.parametrize("chunk_read_limit", [240])  # [0, 240, 102400000])
+@pytest.mark.parametrize("pass_read_limit", [240])  # [0, 240, 102400000])
+@pytest.mark.parametrize(
+    "nrows,skip_rows",
+    [
+        # These are OK
+        # Note: there is a bug if we read the buffer with read_parquet_chunked
+        # before cudf.read_parquet, where the cudf.read_parquet call will
+        # produce an invalid DF
+        (0, 0),
+        (1000, 0),
+        (0, 1000),
+        # segfault in the chunked parquet reader (invalid device ordinal)
+        (1000, 10000),
```

Comment:
I'm still getting an invalid device ordinal error here.

Reply:
Working on this. Meanwhile, I noticed that if you read parquet with `cudf.read_parquet(numrows=.., skiprows=..)` the output columns look ok but the index column doesn't look right, hence you are getting a segfault when comparing the output with the original. Though, if you set `index=True` when `df.to_parquet()` then this goes away. It looks like the index column needs to be handled as well in Python/Cython layer when `index=None or False`.
```
# original table (nrows = 5, skip_rows=1)
    a    b
0   1   av
1   2   qw
2   3   hi
3   4  xyz
4   1   av
5   2   qw
6   3   hi
7   4  xyz
8   1   av
9   2   qw
10  3   hi
11  4  xyz
12  1   av
13  2   qw
14  3   hi
15  4  xyz
16  1   av
17  2   qw
18  3   hi
19  4  xyz

# got           
0         
1         
2         
3         
4         
5         
6         
7         
8         
9         
10        
11        
12        
13        
14  a    b
15  2   qw
16  3   hi
17  4  xyz
18  1   av
19  2   qw

#expected
   a    b
0  2   qw
1  3   hi
2  4  xyz
3  1   av
4  2   qw
```

Reply:
Hi @galipremsagar @vyasr, if possible, could you please comment on if this is libcudf issue or if it should be handled in Cython/Python.

Reply:
This needs to be handled in cython/python layer.

Reply:
Thanks for finding this bug!

Looks like there is an issue processing of the range index metadata afterwards, taking a look!

Reply:
This should be fixed now!

Reply:
@lithomas1  For the chunked parquet reader, unfortunately it looks like the segfault/invalid device ordinal problem isn't trivial so I would recommend not adding it to it until it's solved (in a follow up PR) and keep this PR moving.

Reply:
Please open an issue to do that for now. Thank you!

Reply:
Let's keep track of this in #16195 - I added a comment there.